extensions/wikia/PortableInfobox/controllers/ApiQueryPortableInfobox.class.php

```diff
@@ -41,4 +59,36 @@ protected function runOnPageSet( ApiPageSet $pageSet ) {
 			}
 		}
 	}
+
+	/**
+	 * @desc From the template string with removed <includeonly> tags, creates an array of
+	 * strings containing only infoboxes. All template content which is not an infobox is removed.
+	 *
+	 * @param $text string Content of template which uses the <includeonly> tags
+	 * @return array of striped infoboxes ready to parse
+	 */
+	protected function processTemplate( $text ) {
```

Comment:
Are you sure this preg_match_all("/<infobox.*>.*<\/infobox>/sU", $input_lines, $output_array); will be slower? If yes by how much? Maybe it's worth to use it in favor of readability?